src/main/java/org/folio/helper/BindHelper.java

```diff
@@ -93,19 +116,23 @@ private Map<String, List<Piece>> updateTitleWithBindItems(Map<String, List<Piece
                                                             RequestContext requestContext) {
     piecesByPoLineIds.forEach((poLineId, pieces) -> {
       List<String> itemIds = pieces.stream().map(Piece::getItemId).distinct().toList();
-      titlesService.getTitlesByPoLineIds(List.of(poLineId), requestContext)
-        .map(titles -> {
-            if (titles.containsKey(poLineId) && !titles.get(poLineId).isEmpty()) {
-              Title title = titles.get(poLineId).get(0).withBindItemIds(itemIds);
-              titlesService.saveTitle(title, requestContext);
-            }
-            return titles;
-          }
-        );
+      titlesService.getTitlesByQuery(String.format(TITLE_BY_POLINE_QUERY, poLineId), requestContext)
```

Comment:
current query will return titles by title id, because of id==%s
If you want to return by poLineId query should be like this: poLineId==%s

Reply:
good catch, thanks, fixed